source/common/router/config_impl.h

```diff
@@ -344,6 +345,83 @@ class RouteTracingImpl : public RouteTracing {
   envoy::type::FractionalPercent overall_sampling_;
 };
 
+/**
+ * Base implementation for path rewriters
+ */
+class PathRewriter {
+public:
+  virtual bool apply() const = 0;
+  virtual std::string rewrite(std::string path, const std::string& matched_path,
+                              bool case_sensitive) const = 0;
+  virtual ~PathRewriter(){};
+};
+
+/**
+ * Path rewrite through prefix replacement
+ */
+class PrefixPathRewriter : public PathRewriter {
+public:
+  PrefixPathRewriter(std::string prefix) : PathRewriter(), prefix_(prefix) {}
+
+  bool apply() const { return !prefix_.empty(); }
+
+  std::string rewrite(std::string path, const std::string& matched_path,
+                      bool case_sensitive) const {
+    ASSERT(case_sensitive ? absl::StartsWith(path, matched_path)
+                          : absl::StartsWithIgnoreCase(path, matched_path));
+    return path.replace(0, matched_path.size(), prefix_);
+  }
+
+  ~PrefixPathRewriter(){};
+
+private:
+  std::string prefix_;
+};
+
+/**
+ * Path rewrite via regex
+ */
+class RegexPathRewriter : public PathRewriter {
+public:
+  RegexPathRewriter(std::string pattern, std::string substitution)
+      : PathRewriter(), pattern_(pattern), substitution_(substitution) {}
+
+  bool apply() const { return !substitution_.empty(); }
+
+  std::string rewrite(std::string path, const std::string& matched_path,
+                      bool case_sensitive) const {
+    ASSERT(case_sensitive ? absl::StartsWith(path, matched_path)
+                          : absl::StartsWithIgnoreCase(path, matched_path));
+    return std::regex_replace(matched_path, pattern_, substitution_);
```

Comment:
We don't allow `std::regex` to be used going forward, see https://github.com/envoyproxy/envoy/pull/7878.

Reply:
It it enough to be passing in a type of RE2 or is std::regex_replace itself part of the problem?

Reply:
Yes, please no std::regex of any kind on the data path.